taskcluster/ac_taskgraph/transforms/build.py

```diff
@@ -0,0 +1,66 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+from copy import deepcopy
+from six import text_type
+
+from taskgraph.transforms.base import TransformSequence
+from taskgraph.util.python_path import find_object
+from taskgraph import MAX_DEPENDENCIES
+
+
+transforms = TransformSequence()
+
+
+@transforms.add
+def handle_default_gradle_commands(config, tasks):
+    for task in tasks:
+        if task.pop('include-default-gradle-commands'):
+            task['run']['gradlew'] = [
+                ':{component}:assemble',
+                ':{component}:assembleAndroidTest',
+                ':{component}:test',
+                ':{component}:{lint_task}',
+            ]
+
+        yield task
+
+
+@transforms.add
+def handle_coverage(config, tasks):
+    for task in tasks:
+        if task.pop('include-coverage'):
+            task['run']['gradlew'].insert(0, '-Pcoverage')
+
+        yield task
+
+
+@transforms.add
+def format_component_name_and_lint(config, tasks):
+    for task in tasks:
+        lint_task = 'lintRelease' if task.pop("lint-release") else 'lint'
+
+        for field in ('description', 'run.gradlew', 'treeherder.symbol'):
+            _format_recursively(task, field, component=task["name"], lint_task=lint_task)
+
+        yield task
+
+
+def _format_recursively(object, field, **format_kwargs):
```

Comment:
This is called `_format_recursively`, but it doesn't recurse :thinking:

Reply:
Yeah, I couldn't find a good name yesterday. I just changed it to `_deep_format()` which looks more accurate to me. Please let me know if you have a better idea 🙂